controllers/openshift_controller.go

```diff
@@ -1184,51 +1184,10 @@ func (r *KataConfigOpenShiftReconciler) processKataConfigDeleteRequest() (ctrl.R
 		_ = r.disablePeerPods()
 
 		// Handle podvm image deletion
-		// Since we want to declaratively reach the final state, we need to reconcile when there are errors
-		// as we want the system to give a chance of fixing the error.
-		// For cases we don't want to reconcile, ie for ImageDeletedSuccessfully and UnsupportedPodVMImageProvider
-		// we should just log the message and let the code continue without explicitly returning from the method
-
-		// Following are returned statuses:
-		// ImageDeletedSuccessfully
-		// UnsupportedPodVMImageProvider
-		// ImageDeletionFailed
-		// RequeueNeeded
-		// ImageDeletionStatusUnknown
-
-		// Handle podvm image deletion
-		status, err := ImageDelete(r.Client)
-		switch status {
-		case ImageDeletedSuccessfully:
-			r.setInProgressConditionToPodVMImageDeleted()
-			r.Log.Info("PodVM Image deleted successfully")
-
-		case UnsupportedPodVMImageProvider:
-			r.setInProgressConditionToPodVMImageUnsupportedProvider()
-			r.Log.Info("unsupported cloud provider, skipping image deletion")
-
-		case RequeueNeeded:
-			r.setInProgressConditionToPodVMImageDeleting()
-			return ctrl.Result{Requeue: true, RequeueAfter: 15 * time.Second}, err
-
-		case ImageDeletionFailed:
-			r.setInProgressConditionToPodVMImageDeletionFailed()
-			if err != nil {
-				// We requeue only if there is an error.
-				return ctrl.Result{Requeue: true, RequeueAfter: 15 * time.Second}, err
-			}
-			// If there's no error, log and continue
-			r.Log.Info("Image deletion failed. Check logs for more details")
-
-		case ImageDeletionStatusUnknown:
-			r.setInProgressConditionToPodVMImageDeletionUnknown()
+		err := r.deletePodVMImage()
+		if err != nil {
```

Comment:
We can get there from the `RequeueNeeded` branch in `deletePodVMImage()` with `err` being `nil` and we won't requeue... `deletePodVMImage()` should return `(*ctrl.Result, error)` and you should test either one being not `nil`.